File: 0190_reverse_bits/reverse_bits.cpp
#include <iostream>

using namespace std;

uint32_t reverseBits(uint32_t n)
{
    uint32_t res = 0;
    for (uint32_t i = 31; n; i--, n >>= 1)
        res += (n & 1) << i;
    return res;
}

File: 0190_reverse_bits/reverse_bits.go
package main

func reverseBits(num uint32) uint32 {
	res := uint32(0)
	i := uint32(31)
	for num != 0 {
		res += (num & 1) << i
		num >>= 1
		i--
	}
	return res
}

func main() {

}
